src/RileyMaze/userFunction.js

```diff
@@ -0,0 +1,244 @@
+/* exported RMZUserFunction */
+
+const {GObject, Gtk} = imports.gi;
+
+const {Codeview} = imports.codeview;
+
+const FORWARD = 1;
+const UP = 2;
+const DOWN = 3;
+const JUMP = 4;
+const RILEY = {
+    // Keep this in sync with riley class in
+    // hack-toy-apps/com.endlessm.RileyMaze/parameters.js
+    queue: [],
+    forward() {
+        this.queue.push(FORWARD);
```

Comment:
According to the API document, the instruction methods should throw an exception if the queue is already full, so we can display that to the player. (I saw that you are checking for >8 instructions below already, but throwing an exception here allows us to show the error dot at the right location.)